### PR TITLE
Movie detail

### PR DESCRIPTION
**Pull request by janegreene**
Added new poros for movie details page to display cast, reviews, and movie info
New API connections made for gathering these details
Functionality in place refactor pending.

### CODE REVIEW

File: app/controllers/movies_controller.rb [hunk 1]
```diff
@@ -51,4 +51,31 @@ def index
       end
     end
   end
+  def show
```
Comment (janegreene) [inline app/controllers/movies_controller.rb:54]: In need of refactor but functions as expected.

File: app/poros/cast.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Cast
+  attr_reader :members
+  def initialize(cast_info)
+    @members = cast_info.first(10).inject({}) do |acc, cast_member|
+      acc[cast_member[:name]] =  cast_member[:character]
+      acc
+    end
+  end
+end
```
Comment (michaeljevans) [inline app/poros/cast.rb:9]: super sweet use of .inject

File: app/poros/movie.rb [hunk 3]
```diff
@@ -1,9 +1,21 @@
...
+    @genres   = get_genres(movie_info)
+    @runtime = movie_info[:runtime] if movie_info[:runtime]
     @vote_avg = movie_info[:vote_average].to_f
+    @overview = movie_info[:overview]
+  end
+
+  def get_genres(movie_info)
+    movie_info[:genres].map { |genre| genre[:name] } if movie_info[:genres]
   end
 end
```
Comment (michaeljevans) [inline app/poros/movie.rb:21]: nice guard clauses, yo

File: app/poros/review.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class Review
+  attr_reader :reviews
+  def initialize(review_info)
+    @reviews = review_info.inject({}) do |acc, review|
+      acc[review[:author]] = review[:content]
+      acc
+    end
+  end
+end
```
Comment (michaeljevans) [inline app/poros/review.rb:9]: i dig

File: spec/features/movies/show_spec.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      within '.cast' do
+          expect(page).to have_selector('.member', count: 10)
+      end
+
+      expect(page).to have_css('.reviews')
+      reviews = find('.reviews').text
+      expect(reviews).not_to be_empty
+    end
+  end
+end
```
Comment (michaeljevans) [inline spec/features/movies/show_spec.rb:56]: very thorough test!

File: app/views/movies/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,34 @@
+<section class='movie-info'>
```
Comment (janegreene) [inline app/views/movies/show.html.erb:1]: Needs styling but all required elements are present.

File: spec/features/movies/show_spec.rb [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  describe 'and see movie details page' do
+    it "contains movie reviews cast and other details" do
+      OmniAuth.config.test_mode = true
+      OmniAuth.config.mock_auth[:google_oauth2] = OmniAuth::AuthHash.new(
+        Faker::Omniauth.google
+      )
+      visit root_path
+      click_button 'Log In with Google'
+
+      visit movies_path(search_terms: 'godfather')
```
Comment (janegreene) [inline spec/features/movies/show_spec.rb:13]: Had to search movie with 10 cast members and a review present. This was also more effective than navigating from top_40 so we could predict the path.

**Comment by michaeljevans** [comment]
As Jane said, functionality is in place; we will be refactoring soon!

**Pull request merged**